snuba/snuba_migrations/generic_metrics/0014_distribution_add_options.py

```diff
@@ -0,0 +1,82 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    storage_set_key = StorageSetKey.GENERIC_METRICS_DISTRIBUTIONS
+
+    local_table_name = "generic_metric_distributions_raw_local"
+    dist_table_name = "generic_metric_distributions_raw_dist"
+
+    columns = [
+        (
+            Column("enable_histogram", UInt(8, MigrationModifiers(default=str("0")))),
+            "granularities",
+        ),
+        (
+            Column(
+                "decasecond_retention_days",
+                UInt(8, MigrationModifiers(default=str("7"))),
+            ),
+            "enable_histogram",
+        ),
+        (
+            Column(
+                "min_retention_days",
+                UInt(8, MigrationModifiers(default=str("30"))),
+            ),
+            "decasecond_retention_days",
+        ),
+        (
+            Column(
+                "hr_retention_days",
+                UInt(8, MigrationModifiers(default=str("retention_days"))),
+            ),
+            "min_retention_days",
+        ),
+        (
+            Column(
+                "day_retention_days",
+                UInt(8, MigrationModifiers(default=str("retention_days"))),
+            ),
+            "hr_retention_days",
+        ),
```

Comment:
If we use adedicated column per retention, we would have to run through this process (that includes the change to the matview) if we wanted to introduce a new granularity. Did you consider having an array of nested structures where you could store pairs of granularity and retention? That would not require structural changes to the table to introduce a new granularity.

Reply:
Also how are we going to disable a granularity entirely? do we set the column to 0? 
It is important we do not rely on a retention = 0 to disable a granularity entirely. even a 0 retention would mean data gets materialized and then dropped.

Reply:
I did consider using an array of nested structures, and did not go forward with it because
- From discussions with Nikhar and Oliver it looks like we are unlikely to introduce new granularity, and if so, not very frequently
- With the nested structures, we still need to run migration on the MatViews since we need to update the integer to seconds mapping

We will not be relying on setting retention to zero. The messages sent from Sentry will have some enum that represents the metric aggregation type. The snuba consumers will read that enum and only insert granularities that should be enabled into the granularities columns.

Reply:
> With the nested structures, we still need to run migration on the MatViews since we need to update the integer to seconds mapping

I am not sure this is accurate:
I am talking about this schema 
```
retentions: Nested(
     granularity: UInt,
     retention: Int
)
```

Each row of that array defines a rule you do not need to change the table to add new rules or remove some

Reply:
I see, but wouldn't the MatView still not understand what granularity maps to in seconds?
I.e.
```sql
toDateTime(multiIf(granularity=0,10,
                                 granularity=1,60,
                                 granularity=2,3600,
                                 granularity=3,86400,-1)
...
as timestamp,
```
Don't we need to run another migration to the MatView to add another condition for `multiIf` even if we use the nested structure?